### ISSUE TITLE
Provide a property for configuring the base path of the Actuator's endpoints when using WebFlux

### ISSUE THREAD

**Issue by ferblaca** [spring-projects/spring-boot#22906]
Spring Boot 2.3.2.RELEASE

The settings for customization of monitoring endpoints described at this [link](https://docs.spring.io/spring-boot/docs/2.3.2.RELEASE/reference/html/production-ready-features.html#production-ready-customizing-management-server-context-path) do not work for Webflux applications.

According to the following configuration:

```
spring:
  webflux:
    base-path: /demowebflux
management:
  server:
    port: 9081
    servlet:
      context-path: /management
  endpoints:
    web:
      base-path: /manage
server:
  port: 9080
```

should display the monitoring at "http://localhost:9081/management/manage" but does so at "http://localhost:9081/manage", seems to disregard the "management.server.servlet.context-path" property.

For this other configuration:
```
spring:
  webflux:
    base-path: /demowebflux
management:
  endpoints:
    web:
      base-path: /manage
server:
  port: 9080
```
works correctly the endpoint "http://localhost:9080/demowebflux/manage".

attached demo reproducing the problem.

[demoWebflux.zip](https://github.com/spring-projects/spring-boot/files/5062139/demoWebflux.zip)

**Comment by wilkinsona** [comment]
Thanks for the report. `spring.webflux.base-path` is new in Spring Boot 2.3 and when we introduced it, it looks like we forgot to provide an equivalent of `management.server.servlet.context-path`.

**Comment by wilkinsona** [comment]
I wonder if we should provide a single property, something like `management.server.base-path`, that configures the servlet context path or the web flux base path depending on which web stack you're using. We could then deprecate `management.server.servlet.context-path`.

**Comment by ferblaca** [comment]
Yes, I think that would be a good solution, both for servlet and webflux

**Comment by ferblaca** [comment]
Hi @philwebb!! the issue has changed from 'bug' to 'enhancement' and will be resolved for Spring Boot version 2.4.x, but the milestone 2.4.x has no release date. 

Is it impossible for it to be included in 2.3.x? Sorry for the insistence, but this problem is blocking our deployment.

do you have an approximate date for the milestone 2.4.x?

**Comment by wilkinsona** [comment]
Spring Boot 2.4.0 is scheduled for 12 November. The issue was changed from a bug to an enhancement as the team quite rightly pointed out to me that the problem is due to a missing feature rather than a faulty implementation.
